### PR TITLE
Support alternate Drupal distributions

### LINKED ISSUE #2: Explore options to install a distro instead of Drupal core
Let's explore options to allow popular distros, like Panopoly and OpenAtrium, to be used as the base installation instead of a vanilla copy of Drupal core.

### PR DESCRIPTION
**Pull request by grayside**
The changes below are a fairly significant refactoring of the code to allow individual distributions to take ownership of relevant options presented to the user and the specific operations used to assemble the distribution.

As a case, we have "vanilla" Drupal and Open Atrium in this change. As far as I can tell, both are correctly setting up the codebase and the main Grunt process works fine.
- Drupal 8 is broken by #10 
- Drupal 7 successfully ran with `grunt drush:liteinstall && grunt serve`
- Atrium through an undefined index warning on the "quickstart" variable, and then hit a fatal SQL exception:

```
exception 'PDOException' with message 'SQLSTATE[HY000]: General error: 1 table "defaultconfig_optionals" has more than one primary key' in                              [error]
[...]/tools/hold/build/html/includes/database/sqlite/database.inc:238
Stack trace:
```

This fixes #2

@arithmetric 
@mike-potter

### CODE REVIEW

File: app/distros/index.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+var requireDirectory = require('require-directory');
```
Comment (arithmetric) [inline app/distros/index.js:1]: This module needs to be added to the package.json.

**Comment by arithmetric** [comment]
When I run this and choose any version of Drupal or Atrium, I get the following error:

```
? Which Drupal distribution would you like to use? Drupal
? Which version of Drupal would you like to use? Drupal 7

Ok, I'm going to start assembling this project...

events.js:72
        throw er; // Unhandled 'error' event
              ^
TypeError: Cannot read property 'dependencies' of undefined
    at module.exports.yeoman.generators.Base.extend.writing.packageJson (/Users/Turgeon/work/automation/grush/generator-gadget/app/index.js:120:12)
    at /Users/Turgeon/work/automation/grush/generator-gadget/node_modules/yeoman-generator/lib/base.js:409:16
    at processImmediate [as _immediateCallback] (timers.js:363:15)
```

File: app/index.js [hunk 2]
```diff
@@ -42,69 +59,31 @@ module.exports = yeoman.generators.Base.extend({
...
-          self.log.error('\nAn error occurred while installing Grunt Drupal Tasks.\n');
-          process.exit(1);
-        }
-
-        self.log('\nInstalled version ' + chalk.red(obj.version) + ' of Grunt Drupal Tasks.\n');
-        done();
-      });
-    });
+  installGDT: function() {
+    //require('./gdt')(this).install();
```
Comment (arithmetric) [inline app/index.js]: Maybe this is why there is no package.json?

**Comment by arithmetric** [comment]
Overall, this looks like a good direction. :+1:

I've noted a few issues when testing it.

**Comment by grayside** [comment]
1. Added require-directory to package.json
2. Uncommented gdt installation
3. Standardized minor formatting and default details of makefiles.
4. Move openatrium project.make template to templates/ for reuse across D.O distros.
5. Added test coverage to verify compliance of distro plugins.
6. Adjusted existing test to match new option parameter names (could not get this test to run as it seems to have some specific environment requirements I'm not aware of)

This change does not fix the openatrium DB error noted in the description above, but I think that should be addressed as a follow-up.

**Pull request merged**